engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -233,6 +242,12 @@ public ExtendedSearchIntegrator getExtendedSearchIntegrator() {
 		return extendedIntegrator;
 	}
 
+	/**
+	 * Returns the names of the projection constants (beginning with {@link #HSEARCH_PROJECTION_FIELD_PREFIX}) supported
+	 * by a specific implementation.
+	 */
+	protected abstract Set<String> getSupportedProjectionConstants();
```

Comment:
Btw. things were still functioning correctly: We'd not consider these two as projection constants, letting them pass through, leaving it to the specific impl to deal with them.